formats/log/note.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"fmt"
+
+	"golang.org/x/mod/sumdb/note"
+)
+
+// CheckpointParser is responsible for parsing checkpoints for a specific log.
+// Any valid note must have been signed by all verifiers known by this parser.
+type CheckpointParser struct {
+	logVerifier       note.Verifier
+	logID             string
+	otherVerifiers    note.Verifiers
+	requiredOtherSigs int
+}
+
+// NewCheckpointParser creates a parser for checkpoints that must be signed by the log
+// which is verified by logVKey, and *all* of the other verifiers under otherVerifiers.
```

Comment:
The `all` other verifiers is surprising, I'd expected to see something like "MUST be signed by the log, MAY be cosigned by a subset of these witnesses" and either leave the N of M check to the caller of `Parse` (via the returned `Note.Sigs`) or pass in the N to that call?

Reply:
Take a look at the new API and see what you think.